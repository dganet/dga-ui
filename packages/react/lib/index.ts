import { ButtonComponent } from "./components/Button/ButtonComponent";

export {
  ButtonComponent as Button,
}