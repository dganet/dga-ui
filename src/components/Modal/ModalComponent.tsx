import { useTheme } from '../../hooks/useTheme'
import React, { memo, useEffect } from 'react'
import ReactDOM from 'react-dom'
import { AiOutlineClose } from 'react-icons/ai'
import {
    Container,
    ModalContainer,
    ModalHeader,
    ModalRegularButton,
    ModalRegularContainer,
    ModalRegularContent
} from './ModalComponent.styles'
import { ModalProps } from './ModalComponent.types'
import Button from '../Button/ButtonComponent'

const ModalDOM = ({ children }: { children: React.ReactNode }) => {
    if (typeof window !== 'undefined') {
        const modalRoot =
            document.querySelector('modal-root') ||
            document.createElement('div')
        modalRoot.setAttribute('id', 'modal-root')

        useEffect(() => {
            document.body.appendChild(modalRoot)

            return () => {
                document.body.removeChild(modalRoot)
            }
        }, [modalRoot])

        return ReactDOM.createPortal(children, modalRoot)
    }
    return null
}

const ModalComponent = ({
    children,
    title,
    isOpen = false,
    handleClose
}: ModalProps) => {
    const theme = useTheme()
    return (
        <ModalDOM>
            <Container theme={theme} isOpen={isOpen}>
                <ModalContainer theme={theme}>
                    <ModalHeader theme={theme} hasTitle={!!title}>
                        {title}
                        <div onClick={handleClose}>
                            <AiOutlineClose />
                        </div>
                    </ModalHeader>
                    {children}
                </ModalContainer>
            </Container>
        </ModalDOM>
    )
}

export const ModalRegular = ({ children, ...rest }: ModalProps) => {
    const theme = useTheme()
    return (
        <ModalComponent {...rest}>
            <ModalRegularContainer theme={theme}>
                <ModalRegularContent>{children}</ModalRegularContent>
                <ModalRegularButton>
                    <Button
                        style={{ marginRight: '10px' }}
                        buttonSize="medium"
                        buttonType="white"
                    >
                        Cancel
                    </Button>
                    <Button buttonSize="medium">Confirm</Button>
                </ModalRegularButton>
            </ModalRegularContainer>
        </ModalComponent>
    )
}

export default memo(ModalComponent)
