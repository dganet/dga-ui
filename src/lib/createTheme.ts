import { defaultTheme } from '../styles/defaultTheme'

export const createTheme = (theme: typeof defaultTheme) => {
    return {
        ...defaultTheme,
        colors: {
            ...defaultTheme.colors,
            ...theme.colors
        },

        name: theme.name || defaultTheme.name
    } as typeof defaultTheme
}
