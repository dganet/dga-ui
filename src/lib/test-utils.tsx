import { render, RenderOptions } from '@testing-library/react'
import React, { ReactElement, FC } from 'react'
import { defaultTheme } from '../styles/defaultTheme'
import { CustomThemeProvider } from '../hooks/useTheme/ThemeContext'

const Providers: FC = ({ children }) => {
    return (
        <CustomThemeProvider theme={defaultTheme}>
            {children}
        </CustomThemeProvider>
    )
}

/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
const customRender = (
    ui: ReactElement,
    options?: Omit<RenderOptions, 'queries'>
) => render(ui, { wrapper: Providers, ...options })

// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }
