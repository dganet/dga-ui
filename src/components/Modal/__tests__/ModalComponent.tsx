import { render, screen } from '../../../lib/test-utils'

import ModalComponent from '../ModalComponent'

describe('<ModalComponent />', () => {
    it('should render the component', () => {
        let isOpen = true
        const { container } = render(
            <ModalComponent
                title=""
                isOpen={isOpen}
                handleClose={() => (isOpen = false)}
            >
                Teste
            </ModalComponent>
        )

        expect(container.firstChild).toMatchSnapshot()
    })
})
