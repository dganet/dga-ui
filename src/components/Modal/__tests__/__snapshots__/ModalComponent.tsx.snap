// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ModalComponent /> should render the component 1`] = `null`;
